src/Models/Market.php

```diff
@@ -61,6 +83,15 @@ protected static function boot()
         });
     }
 
+    public function updatingSlug()
+    {
+        $this->slug = $this->slug ?: Str::slug($this->city);
+        $invalidSlugs = config('locations.invalid_slugs') ?? [];
+        if (in_array($this->slug, $invalidSlugs)) {
+            $this->slug = Str::slug("{$this->slug}-{$this->state->slug}");
+        }
+    }
+
```

Comment:
With all slugs now being unique via the page model, this should be removed (along with the defaults in the `config/locations.php` file